c10/util/ThreadLocalDebugInfo.cpp

```diff
@@ -51,6 +51,15 @@ std::shared_ptr<DebugInfoBase> ThreadLocalDebugInfo::_pop(DebugInfoKind kind) {
   return res->info_;
 }
 
+/* static */
+std::shared_ptr<DebugInfoBase> ThreadLocalDebugInfo::_peek(DebugInfoKind kind) {
+  TORCH_CHECK(
+    debug_info && debug_info->kind_ == kind,
+    "Expected debug info of type ", (size_t) kind);
+    auto res = debug_info;
+    return res->info_;
```

Comment:
BTW, why creating a copy of `debug_info` instead of directly return `debug_info->info_`?

Reply:
Was just following the above code, but it can just be `debug_info->info_` indeed.